src/engine/ruleset/decoders/fortinet/fortinet-firewall.yml

```diff
@@ -0,0 +1,498 @@
+      - fortinet.firewall.server: $~tmp.server
+      - fortinet.firewall.version: $~tmp.version
+      - fortinet.firewall.applist: $~tmp.applist
+      - fortinet.firewall.app: $network.application
+      - fortinet.firewall.rcvddelta: $~tmp.rcvddelta
+      - fortinet.firewall.sentdelta: $~tmp.sentdelta
+      - fortinet.firewall.vwlid: $~tmp.vwlid
+      - fortinet.firewall.identifier: $~tmp.identifier
+      - fortinet.firewall.appid: $~tmp.appid
+      - fortinet.firewall.apprisk: $~tmp.apprisk
+      - fortinet.firewall.appact: $~tmp.appact
+      - fortinet.firewall.vwlquality: $~tmp.vwlquality
+      - fortinet.firewall.wanin: $~tmp.wanin
+      - fortinet.firewall.wanout: $~tmp.wanout
+      - fortinet.firewall.lanin: $~tmp.lanin
+      - fortinet.firewall.lanout: $~tmp.lanout
+      - fortinet.firewall.utmaction: $~tmp.utmaction
+      - fortinet.firewall.countweb: $~tmp.countweb
+      - fortinet.firewall.countapp: $~tmp.countapp
+      - fortinet.firewall.countips: $~tmp.countips
+      - fortinet.firewall.utmref: $~tmp.utmref
+      - fortinet.firewall.countav: $~tmp.countav
+      - fortinet.firewall.osname: $~tmp.osname
+      - fortinet.firewall.mastersrcmac: $~tmp.mastersrcmac
+      - fortinet.firewall.srcmac: $~tmp.srcmac
+      - fortinet.firewall.srcserver: $~tmp.srcserver
+      - fortinet.firewall.dstuuid: $~tmp.dstuuid
+      - fortinet.firewall.srcuuid: $~tmp.srcuuid
+      - fortinet.firewall.devcategory: $~tmp.devcategory
+      - fortinet.firewall.devtype: $~tmp.devtype
+      - fortinet.firewall.countdlp: $~tmp.countdlp
+      - fortinet.firewall.cat: $~tmp.cat
+      - fortinet.firewall.eventtype: $~tmp.eventtype
+      - fortinet.firewall.method: $~tmp.method
+      - fortinet.firewall.reqtype: $~tmp.reqtype
+      - fortinet.firewall.incidentserialno: $~tmp.incidentserialno
+      - fortinet.firewall.qtypeval: $~tmp.qtypeval
+      - fortinet.firewall.analyticscksum: $~tmp.analyticscksum
+      - fortinet.firewall.analyticssubmit: $~tmp.analyticssubmit
+      - fortinet.firewall.quarskip: $~tmp.quarskip
+      - fortinet.firewall.virus: $~tmp.virus
+      - fortinet.firewall.virusid: $~tmp.virusid
+      - fortinet.firewall.attack: $~tmp.attack
+      - fortinet.firewall.attackid: $~tmp.attackid
+      - fortinet.firewall.icmpcode: $~tmp.icmpcode
+      - fortinet.firewall.icmpid: $~tmp.icmpid
+      - fortinet.firewall.icmptype: $~tmp.icmptype
+      - fortinet.firewall.policytype: $~tmp.policytype
+      - fortinet.firewall.dlpextra: $~tmp.dlpextra
+      - fortinet.firewall.epoch: $~tmp.epoch
+      - fortinet.firewall.filtercat: $~tmp.filtercat
+      - fortinet.firewall.filteridx: $~tmp.filteridx
+      - fortinet.firewall.filtertype: $~tmp.filtertype
+      - fortinet.firewall.channeltype: $~tmp.channeltype
+      - fortinet.firewall.login: $~tmp.login
+      - fortinet.firewall.certhash: $~tmp.certhash
+      - fortinet.firewall.quarskip: $~tmp.quarskip
```

Comment:
If all the fields are called the same, maybe you could just do an `ef_merge` of (`fortinet.firewall` and `~tmp`) instead of copying so many values. Please, check that out.